src/kanata/linux.rs

```diff
@@ -51,15 +51,82 @@ impl Kanata {
 
                 check_for_exit(&key_event);
 
-                // Check if this keycode is mapped in the configuration. If it hasn't been mapped, send
-                // it immediately.
-                if !MAPPED_KEYS.lock().contains(&key_event.code) {
-                    let mut kanata = kanata.lock();
-                    kanata
-                        .kbd_out
-                        .write_key(key_event.code, key_event.value)
-                        .map_err(|e| anyhow!("failed write key: {}", e))?;
-                    continue;
+                let KeyEvent { code, value } = key_event;
+
+                if value == KeyValue::Tap {
+                    // Scroll event for sure. Only scroll events produce Tap.
+
+                    let direction: MWheelDirection = code.try_into().unwrap();
+
+                    match code
+                        .try_into()
+                        .expect("scroll event OsCode should not fail this")
+                    {
+                        ScrollEventKind::Standard => {
+                            if kanata.lock().scroll_wheel_mapped {
```

Comment:
Okay, let me explain. There is a weird buggy behavior (at least on linux) particularly in apps that care about hi-res scroll (e.g. vscode or chrome), when an input device alternates between sending the kind or report that contains a hi-res scroll e.g,
```
Event: time 1698616141.279432, type 2 (EV_REL), code 8 (REL_WHEEL), value -1
Event: time 1698616141.279432, type 2 (EV_REL), code 11 (REL_WHEEL_HI_RES), value -120
Event: time 1698616141.279432, -------------- SYN_REPORT ------------
```
and a report that without hi-res scroll e.g.,
```
Event: time 1698616141.279432, type 2 (EV_REL), code 8 (REL_WHEEL), value -1
Event: time 1698616141.279432, -------------- SYN_REPORT ------------
```
depending on which kind of report came first (with hi-res or without) there will be a different bug, either: 
1. the reports without hi-res scroll will simply don't scroll the page and the reports that contain hi-res scroll will scroll 1 notch (as intended)
2. or the reports without hi-res scroll will scroll 1 notch (as intended), but the report that contain hi-res scroll will scroll 2 notches

In the past https://github.com/jtroo/kanata/issues/395 was somehow resolved by #398 and I'm still confused why. But since #398 I haven't seen any complaints about scroll on linux, so I guess this was done correctly.

Reply:
The top-most priority is to ensure that there's no situation when above 2 types of scroll reports are mixed. When we have info that user remapped scroll (`scroll_wheel_mapped == true`), it means that it could be remapped to `_` or `mouse-wheel` or in general `KbdOut::scroll`, which ALWAYS outputs `REL_WHEEL_HI_RES` + `REL_HWHEEL` in a single report. So to not mix types of reports, whenever we would passthough a scroll event (i.e. whenever `MAPPED_KEYS.lock().contains(&code) == false`) kanata instead converts every `REL_HWHEEL` event to a pair of events `REL_WHEEL_HI_RES` + `REL_HWHEEL` and because we already output `REL_WHEEL_HI_RES` whenever we come across `REL_HWHEEL`, `REL_WHEEL_HI_RES` input event is dropped.

Next priority is to ensure that if somebody didn't remap scroll event, theres no way kanata breaks scroll in anyway. One way to ensure that is to pass-through every event related to scrolling. This is best done by using `write_raw`.

I'm begging you, boot linux and play around with the code. This shit is pretty hard to explain let alone fully grasp with nothing but theory.

Reply:
This is probably worth putting in the code as documentation in comments 🙂. Without the context, the two checks look redundant at a glance, and they might get accidentally removed at some point.

Reply:
okay

Reply:
I've already left a comment (lines 70-77) attempting to explain some of this. Feel free to change this comment as you wish and add other comments with additional explanations.